### DISCUSSION TITLE
Inverse wrap field

### DISCUSSION THREAD

**Discussion by maxublinsk** [kschan0214/sepia#69] [Q&A]
Hello!
I have question about SEPIA, Subcortical structure segmentation, Approach 2. Where can I get T1w to atlas T1w template inverse wrap field? GRE to T1w rigid-body transformation matrix file and T1w to atlas T1w template affine transformation matrix file I can obtain using FSL FLIRT, for example. But I didn't find how to obtain inverse wrap field (nii.gz).

**Comment by kschan0214** [comment]
Hi, 

The current implementation for subcortical segmentation is built on nonlinear image registration using [ANTs](https://github.com/ANTsX/ANTs) (i.e. GRE->T1w: rigid; T1w->Atlas template: Affine+nonlinear(SyN)). Therefore, it is recommended to obtain the required transformation files (rigid, affine transform matrices and (inverse) warp field) using ANTs (which is what Approach 1 does). 

Nonetheless, if you have FSL's FLIRT transformation matrix, you might convert the files to ANTs compatible format (e.g. ITK) using software like [c3d_affine_tool](http://www.itksnap.org/pmwiki/pmwiki.php?n=Downloads.C3D). If you use FSL for image registration, it is possible to use FNIRT to obtain the warp field (and inverse warp I believe). However, I don't have any experience on applying the FSL's warp field directly on ANTs pipeline (antsregistration) so I can't guarantee they are compatible :/. 

Kwok
